src/spells.cpp

```diff
@@ -235,17 +235,15 @@ uint32_t Spells::getInstantSpellCount(const Player* player) const
 	return count;
 }
 
-InstantSpell* Spells::getInstantSpellByIndex(const Player* player, uint32_t index)
+InstantSpell* Spells::getInstantSpellById(uint32_t spellId)
 {
 	uint32_t count = 0;
 	for (const auto& it : instants) {
 		InstantSpell* instantSpell = it.second;
-		if (instantSpell->canCast(player)) {
-			if (count == index) {
-				return instantSpell;
-			}
-			++count;
+		if (count == spellId) {
+			return instantSpell;
 		}
+		++count;
 	}
 	return nullptr;
```

Comment:
This function is now pretty ugly. Please make it like:

```cpp
auto it = std::advance(instants.begin(), index);
if (it != instants.end()) {
  return it->second;
}
return nullptr;
```

It's much shorter and cleaner